sql/src/test/java/io/crate/metadata/RoutingTest.java

```diff
@@ -64,4 +64,44 @@ public void testStreamingWithoutLocations() throws Exception {
         Routing routing2 = Routing.fromStream(in);
         assertThat(routing1.locations(), is(routing2.locations()));
     }
+
+    @Test
+    public void testUpdateLocations() {
+        Map<String, Map<String, List<Integer>>> exitingLocations = new TreeMap<>();
+        Map<String, List<Integer>> indexMapExisting = new TreeMap<>();
+        indexMapExisting.put("index-0", Arrays.asList(1, 2));
+        indexMapExisting.put("index-1", Arrays.asList(1, 2));
+        exitingLocations.put("node-0", indexMapExisting);
+        indexMapExisting = new TreeMap<>();
+        indexMapExisting.put("index-0", Arrays.asList(3, 4));
+        indexMapExisting.put("index-1", Arrays.asList(3, 4));
+        exitingLocations.put("node-1", indexMapExisting);
+
+        Routing routing = new Routing(exitingLocations);
+        routing.updateLocations(exitingLocations);
+        // Same locations must be returned
+        assertThat(exitingLocations, is(routing.locations()));
+
+        Map<String, Map<String, List<Integer>>> newLocations = new TreeMap<>();
+        Map<String, List<Integer>> indexMapNew = new TreeMap<>();
+        indexMapNew.put("index-0", Arrays.asList(3, 4, 5));
+        indexMapNew.put("index-1", Arrays.asList(3, 4, 5));
+        newLocations.put("node-0", indexMapNew);
+        indexMapNew = new TreeMap<>();
+        indexMapNew.put("index-0", Arrays.asList(1, 2, 5));
+        indexMapNew.put("index-1", Arrays.asList(1, 2, 5));
```

Comment:
this is an invalid case, where 5 is two nodes
and aactually the result below is also invalid because of this